backend/src/apiserver/storage/pipeline_store_kubernetes.go

```diff
@@ -0,0 +1,519 @@
+		return err
+	}
+
+	conditionSet := false
+
+	for i := range k8sPipelineVersion.Status.Conditions {
+		condition := &k8sPipelineVersion.Status.Conditions[i]
+
+		if condition.Type == "PipelineVersionStatus" {
+			if condition.Reason == string(status) && condition.Message == condition.Reason {
+				return nil
+			}
+
+			condition.Reason = string(status)
+			condition.Message = string(status)
+			condition.Status = metav1.ConditionTrue
+
+			conditionSet = true
+
+			break
+		}
+	}
+
+	if !conditionSet {
+		k8sPipelineVersion.Status.Conditions = append(k8sPipelineVersion.Status.Conditions, v2beta1.SimplifiedCondition{
+			Type:    "PipelineVersionStatus",
+			Reason:  string(status),
+			Message: string(status),
+			Status:  metav1.ConditionTrue,
+		})
+	}
+
+	err = k.client.Status().Update(context.TODO(), k8sPipelineVersion)
+	if err != nil && k8serrors.IsConflict(err) {
+		return k.UpdatePipelineVersionStatus(pipelineVersionId, status)
+	} else if err != nil {
+		return util.NewInternalServerError(err, "Failed to update the pipeline version status")
+	}
+	return nil
+}
+
+func (k *PipelineStoreKubernetes) CreatePipelineVersion(pipelineVersion *model.PipelineVersion) (*model.PipelineVersion, error) {
+	pipeline, err := k.GetPipeline(pipelineVersion.PipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	return k.createPipelineVersionWithPipeline(context.TODO(), pipeline, pipelineVersion)
+}
+
+func (k *PipelineStoreKubernetes) UpdatePipelineDefaultVersion(pipelineId string, versionId string) error {
+	// Default version was used in KFPv1 and is deprecated. In KFPv2, we do not support this.
+	return util.NewBadRequestError(errors.New("pipeline default version is unsupported"),
+		"pipeline default version is unsupported when storing in Kubernetes")
+}
+
+func (k *PipelineStoreKubernetes) GetLatestPipelineVersion(pipelineId string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO(), pipelineId)
+	if err != nil {
+		return nil, err
+	}
+
+	var latestK8sPipelineVersion *v2beta1.PipelineVersion
+
+	for _, k8sPipelineVersion := range k8sPipelineVersions.Items {
+		if latestK8sPipelineVersion == nil || k8sPipelineVersion.CreationTimestamp.Time.After(latestK8sPipelineVersion.CreationTimestamp.Time) {
+			latestK8sPipelineVersion = &k8sPipelineVersion
+
+			continue
+		}
+	}
+
+	if latestK8sPipelineVersion == nil {
+		return nil, util.NewResourceNotFoundError("PipelineVersion", "Latest")
+	}
+
+	return latestK8sPipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersion(pipelineVersionId string) (*model.PipelineVersion, error) {
+	pipelineVersion, err := k.getK8sPipelineVersion(context.TODO(), pipelineVersionId)
+	if err != nil {
+		return nil, err
+	}
+
+	return pipelineVersion.ToModel()
+}
+
+func (k *PipelineStoreKubernetes) GetPipelineVersionByName(name string) (*model.PipelineVersion, error) {
+	k8sPipelineVersions, err := k.getK8sPipelineVersions(context.TODO(), "")
```

Comment:
It'd be better if you used a field selector to filter by `metadata.name` rather than client side filtering.